app/controllers/webhook_controller.rb

```diff
@@ -55,17 +21,80 @@ def callback
             type: 'text',
             text: event.message['text']
           }
+          p message
           client.reply_message(event['replyToken'], message)
         when Line::Bot::Event::MessageType::Location
           res_text = search_shop(event.message['latitude'], event.message['longitude'])
           message = {
-            type: 'text',
-            text: res_text.to_s
+            "type": "template",
+            "altText": "おすすめのお店情報が届きました！",
+            "template": {
+              "type": "carousel",
+              "columns": res_text,
+              "imageAspectRatio": "rectangle",
+              "imageSize": "cover"
+            }
           }
           client.reply_message(event['replyToken'], message)
         end
       end
     }
     head :ok
   end
+
+
+  private
+  hotpepper_api = "http://webservice.recruit.co.jp/hotpepper/gourmet/v1/"
+
+  def client
+    @client ||= Line::Bot::Client.new { |config|
+      config.channel_secret = ENV["LINE_CHANNEL_SECRET"]
+      config.channel_token = ENV["LINE_CHANNEL_TOKEN"]
+    }
+  end
+
+  
+  def search_shop(lat, lng)
+    uri = URI.parse(hotpepper_api)
+    http = Net::HTTP.new(uri.host, uri.port) 
+    # hotpepper apiのパラメータ
+    # range 3: 1000m以内
+    # genre G001:居酒屋, G002:ダイニングバー・バル
+    # budget B002:2001~3000円, B003:3001~4000円
+    uri.query = URI.encode_www_form({
+        key: ENV["API_KEY"],
+        lat: lat,
+        lng: lng,
+        range: 3,
+        genre: "G001,G002",
+        budget: "B002,B003",
+        order: 4,
+        format: 'json'
+    })
+    # p uri
+    request = Net::HTTP::Get.new(uri)
+    response = http.request(request)
+
+    if response.code == '200'
+      data = JSON.parse(response.body)
+      columns = []
+      data["results"]["shop"].each do |shop|
+        content = {
+          thumbnailImageUrl: shop["photo"]["mobile"]["l"],
+          title: shop["name"],
+          text: shop["address"],
+          actions: [{
+            type: "uri",
+            label: "詳細を見る",
+            uri: shop["urls"]["pc"]
+          }]
+        }
+        columns.push(content)
+      end
+      return columns
```

Comment:
```suggestion
      columns
```

仕様変わらないと思いますが、Rubyでは必要ない場合はreturnを省略するのが慣例的な書き方になると思います。